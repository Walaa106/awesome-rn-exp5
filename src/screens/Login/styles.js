import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  logoWrapper: {
    alignSelf: 'stretch',
    justifyContent: 'center',
  },
  heading: {
    color: 'white',
    fontSize: 25,
  },
  row: {
    flexDirection: 'row',
    flex: 1,
    justifyContent: 'space-between'
  },
  footer: {
    flexDirection: "row",
    flex: 1,
    justifyContent: 'space-between',
    marginTop: 25,
    paddingLeft: 15
  }
});

export default styles;
