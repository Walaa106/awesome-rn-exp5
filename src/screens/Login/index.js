import React, { Component } from 'react';
import { Container, Content, Header, Left, H3, Button, Text, View } from 'native-base';

import styles from './styles';

export default class Login extends Component {
  render() {
    const { t, navigation } = this.props;
    return (
        <Container>
          <Header style={{ paddingBottom: 30 }}>
            <Left><H3> {t('login:property-login')} </H3></Left>
          </Header>
          <Content>
            {this.props.loginForm}
            <View style={styles.footer}>
              <Text
                style={{ alignSelf: 'center', fontSize: 14 }}
                onPress={() => navigation.navigate('Home')}
              > {t('login:forget-password')} </Text>
              <Button block info
                onPress={() => this.props.onLogin()}
                style={{paddingRight: 20, paddingLeft: 20}}>
                <Text>{t('login:login')}</Text>
              </Button>
            </View>
          </Content>
        </Container>
    )}
}
