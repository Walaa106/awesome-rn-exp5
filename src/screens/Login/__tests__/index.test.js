import 'react-native';
import React from 'react';
import Login from '../';
import renderer from 'react-test-renderer';

const onLogin = jest.fn();
const loginForm = React.Component;

it('renders correctly', () => {
  const tree = renderer.create(<Login onLogin={onLogin} loginForm={<loginForm />} t={(translation)=>translation} />).toJSON();
  expect(tree).toMatchSnapshot();
});
