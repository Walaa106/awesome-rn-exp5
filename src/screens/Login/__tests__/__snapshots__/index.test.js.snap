// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F8F8F8",
            "borderBottomColor": "#a7a6ab",
            "borderBottomWidth": 0.5,
            "elevation": 3,
            "flexDirection": "row",
            "height": 64,
            "justifyContent": "center",
            "left": 0,
            "paddingLeft": 6,
            "paddingRight": 10,
            "paddingTop": 18,
            "right": 0,
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
            "top": 0,
          },
          Object {
            "paddingBottom": 30,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontSize": 21,
              "lineHeight": 22,
            }
          }
        >
           
          login:property-login
           
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": undefined,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    viewIsInsideTabBar={false}
  >
    <View>
      <loginForm />
      <View
        style={
          Array [
            Object {},
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 25,
              "paddingLeft": 15,
            },
          ]
        }
      >
        <Text
          onPress={[Function]}
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "alignSelf": "center",
                "fontSize": 14,
              },
            ]
          }
          uppercase={false}
        >
           
          login:forget-password
           
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#62B1F6",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 6,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
            }
          }
        >
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#fff",
                "fontFamily": "System",
                "fontSize": 16.5,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              }
            }
            uppercase={false}
          >
            login:login
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
